packages/components/src/internal/util/measurement.ts

```diff
@@ -195,13 +195,16 @@ export function areUnitsCompatible(unitAStr: string, unitBStr: string) {
     return unitA.baseUnit == unitB.baseUnit;
 }
 
-export function getMetricUnitOptions(): any[] {
+export function getMetricUnitOptions(metricUnit?: string): any[] {
+    if (metricUnit) {
+        return getAltMetricUnitOptions(metricUnit, true);
+    }
```

Comment:
Nice, glad to see we already had `getAltMetricUnitOptions` for this filtering case. Unrelated to this PR change, but it seems these two methods could be consolidated to handle both the filter by param and return all cases. Also, it is a little odd that one of them checks for `value.label === 'unit'` in the if statement and the other uses `!showLongLabel`, any reason for that distinction?

Reply:
Yes, I think both checks are valid. In the case of `value.label === 'unit'`, we're preventing showing a redundant label on this one case since `unit` is already a word that needs no further explanation. In the case of `!showLongLabel`, we are preventing showing the long label for all the menu options. I might argue that the long label is superfluous in any case, but am going to opt not to change that at this juncture.

I've merged these two methods together, which should lead to more consistency overall.